### ISSUE TITLE
Add Redis cache for GitHub responses

### ISSUE THREAD

**Issue by mattrosno** [carbon-design-system/carbon-platform#884]
**Summary**

The web app uses a [Node.js cache-manager](https://www.npmjs.com/package/cache-manager) to cache all GitHub responses to prevent the web app from getting GitHub API rate limited. These GitHub requests vary from fetching repository information, traversing GitHub trees, to primarily fetching GitHub file content to read carbon.yml files and MDX files for web page content.

We're currently using a file system store engine (https://www.npmjs.com/package/cache-manager-fs-hash), which was chosen during v1 preview prototyping as a quick and easy way to prevent GitHub API rate limits.

As we index more content (asset documentation pages), each GitHub request per MDX file adds to the amount of cached responses. Our current build pipeline does not persist cached files written to disk between builds.

As the amount of indexed content grows, so does the risk of our production GitHub access tokens getting rate limited which could leave us with a production environment void of all content.

**Potential solution**

To mitigate that risk, until we can create proper back-end services for this, what if we added Redis stores for staging and production? It looks like we could use a Redis store listed here: https://github.com/BryanDonovan/node-cache-manager#store-engines

Those Redis store engines _should_ share the same API as `cache-manager-fs-hash`, so that _should_ require minimal web app changes.

We could keep the [1 hour TTL](https://github.com/carbon-design-system/carbon-platform/blob/main/services/web-app/lib/file-cache.js#L24) for cached responses, so for Next.js ISR pages, content from GitHub would be at most 1 hour stale.

With a Redis cache, our production builds should be faster because there's a greater chance of a cache hit.

We could use a Redis cache for local development too, or for simplicity, keep that with the current file system cache.

**TL;DR:**

In the event that our production environment gets rate limited by GitHub, because the Redis cache would be persisted, we wouldn't lose web app content between deploys. Also, it would be neat to see how many keys would get cached in Redis, to give us a better idea of to-be infrastructure needs to manage GitHub content.

**Comment by mattrosno** [comment]
Sub issues could be:

- [ ] Provision services
- [ ] Swap out engine stores
- [ ] What's this mean for CI/CD?

Does this mean all pages would need to be ISR so we don't have to depend on an external service for production builds?

**Comment by mattrosno** [comment]
@jdharvey-ibm to migrate this issue to a technical design issue and close this once done.

**Comment by mattrosno** [comment]
Closing as not planned per @jdharvey-ibm's recommendation. We don't expect our current content needs will get GitHub API rate limited. We can re-evaluate our caching needs when we have a service in place that provides its GitHub data to the data graph. Assuming we have that, it's unlikely that we'll need a centralized data cache.
